torch/csrc/cuda/shared/nvtx.cpp

```diff
@@ -1,11 +1,41 @@
 #ifdef _WIN32
 #include <wchar.h> // _wgetenv for nvtx
 #endif
+#include <cuda_runtime.h>
 #include <nvToolsExt.h>
 #include <torch/csrc/utils/pybind.h>
 
 namespace torch::cuda::shared {
 
+struct RangeHandle {
+  nvtxRangeId_t id;
+  const char* msg;
+};
+
+static void device_callback_range_end(void* userData) {
+  RangeHandle* handle = ((RangeHandle*)userData);
+  nvtxRangeEnd(handle->id);
+  free((void*)handle->msg);
+  free((void*)handle);
+}
+
+static void device_nvtxRangeEnd(void* handle) {
+  cudaLaunchHostFunc(0, device_callback_range_end, handle);
```

Comment:
The other gotcha worth mentioning: these methods operate on the default stream. Shall I add support for different streams as well?

Reply:
Added stream parameter as well, now it isn't locked to the default stream only https://github.com/pytorch/pytorch/pull/130429/commits/a3404af19d3554883a7a64daeeb1bf9268a2a976